src/main/java/org/apache/mesos/hdfs/executor/NameNodeExecutor.java

```diff
@@ -106,13 +113,53 @@ public void frameworkMessage(ExecutorDriver driver, byte[] msg) {
       } else {
         deleteFile(nameDir);
         nameDir.mkdirs();
-        runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
-        startProcess(driver, nameNodeTask);
-        startProcess(driver, zkfcNodeTask);
+      }
+      runCommand(driver, nameNodeTask, "bin/hdfs-mesos-namenode " + messageStr);
+      startProcess(driver, nameNodeTask);
+      startProcess(driver, zkfcNodeTask);
+      driver.sendStatusUpdate(TaskStatus.newBuilder()
+          .setTaskId(nameNodeTask.taskInfo.getTaskId())
+          .setState(TaskState.TASK_RUNNING)
+          .setMessage(messageStr)
+          .build());
+    }
+  }
+
+  /**
+   * This class is designed to help launch the secondary namenode in its bootstrap phase. The node
+   * needs both name nodes to be available as DNS entries. In Mesos-DNS, this takes an uncertain
+   * amount of time, so we must schedule it asynchronously.
+   */
```

Comment:
You are already doing the check in the scheduler so there's no reason to have this code here. You will not get to the bootstrap code until you are in the proper phase.